### PR TITLE
Fix the container test.

### PR DESCRIPTION
**Pull request by pgavlin**
Build the test on the host, then mount it and run it in the container.
This allows us to make use of the usual integration testing framework to
handle running `npm/yarn`, `pip`, etc.

### CODE REVIEW

**Comment by pgavlin** [comment]
I've validated this locally.

**Comment by pgavlin** [comment]
ping @chrsmith

File: tests/containers/containers_test.go [hunk 1]
```diff
@@ -22,11 +22,56 @@ import (
...
+				Dir: e.RootPath,
+			})
+
+			integration.ProgramTest(t, &example)
+		})
+	}
+}
+
+// TestPulumiActionsImage simulates building and running Pulumi programs on the pulumi/actions image.
+func TestPulumiActionsImage(t *testing.T) {
```
Comment (lukehoban) [inline tests/containers/containers_test.go]: It appears we are never running this test - is that right/intentional?

Comment (lukehoban) [reply tests/containers/containers_test.go]: I'll admit this feels a little weird now too - we have some tests that must run outside and some that must run inside.  I don't exactly follow why we wanted these to diverge.

Comment (chrsmith) [reply tests/containers/containers_test.go]: @lukehoban yes, it was intentional.

The `pulumi/pulumi` tests run _within_ the Docker container, and go through the prototypical `pulumi stack init`, `pulumi up`, ... set of scenarios. Those tests are to validate that Pulumi does all the things it needs to do within that container runtime.

The `pulumi/actions` tests cover a different use case. `pulumi/actions` is used for GitHub Actions, and is used for things like CI. So the most common use-case is simply passing the args "up --stack foo/bar". And then the entry point in the Dockerfile will run `yarn install` or `pip`, etc. to get the right dependencies and then run `pulumi`.

So specifically we want those tests to exercise the entry point for the docker container itself.

It's complicated and should be called out in some long comments, but that's the idea anyway.

For expediency Pat went and added just this set of tests. I'll send out a PR to hook up the `pulumi/actions` container, running outside of the Docker container soonish.

**Comment by lukehoban** [comment]
> I'll send out a PR to hook up the pulumi/actions container, running outside of the Docker container soonish.

Got it - great.

**Pull request merged**